peps/pep-0810.rst

```diff
@@ -0,0 +1,1345 @@
+- Less clear and standard than dedicated syntax
+
+**Q: Will this break tools like** ``isort`` **or** ``black``?
+
+A: Tools will need updates to recognize the ``lazy`` keyword, but the changes
+should be minimal since the import structure remains the same. The keyword
+appears at the beginning, making it easy to parse.
+
+**Q: How do I know if a library is compatible with lazy imports?**
+
+A: Most libraries should work fine with lazy imports. Libraries that might have issues:
+
+- Those with essential import-time side effects (registration, monkey-patching)
+- Those that expect specific import ordering
+- Those that modify global state during import
+
+When in doubt, test lazy imports with your specific use cases.
+
+**Q: Can I use lazy imports inside functions?**
+
+A: No, the ``lazy`` keyword is only allowed at module level. For function-level
+lazy loading, use traditional inline imports or move the import to module level
+with ``lazy``.
+
+**Q: What about forwards compatibility with older Python versions?**
+
+A: Use the ``__lazy_modules__`` global for compatibility:
+
+.. code-block:: python
+
+  # Works on Python 3.15+ as lazy, eager on older versions
+  __lazy_modules__ = ['expensive_module', 'expensive_module_2']
+  import expensive_module
+  from expensive_module_2 import MyClass
+
+The ``__lazy_modules__`` attribute is a list of module name strings. When an import
+statement is executed, Python checks if the module name being imported appears in
+``__lazy_modules__``. If it does, the import is treated as if it had the ``lazy``
+keyword (becoming *potentially lazy*). On Python versions before 3.15 that don't
+support lazy imports, the ``__lazy_modules__`` attribute is simply ignored and
+imports proceed eagerly as normal.
+
+This provides a migration path until you can rely on the ``lazy`` keyword. For
+maximum predictability, it's recommended to define ``__lazy_modules__`` once,
+before any imports. But as it is checked on each import, it can be modified between
+``import`` statements.
+
+**Q: How do explicit lazy imports interact with PEP-649/PEP-749**
+
+A: If an annotation is not stringified, then it is a reference to the type, but it
+will only be resolved if the annotation is accessed. In the example below, the ``fake_typing`` module
+is only loaded when the user inspects the ``__annotations__`` dictionary. The ``fake_typing``
+module would also be loaded if the user had leveraged ``inspect.get_annotations()`` or ``getattr`` to access
+the annotations.
+
+.. code-block:: python
+
+  lazy from fake_typing import MyFakeType
+  def foo(x: MyFakeType):
+    pass
+  print(foo.__annotations__)  # Triggers loading the fake_typing module
+
+**Q: How do lazy imports interact with** ``dir()``, ``getattr()``, **and module introspection?**
+
+A: Accessing lazy imports through normal attribute access or ``getattr()`` will trigger
+reification. Calling ``dir()`` on a module will reify all lazy imports in that module
```

Comment:
Why? dir() only shows the names in the namespace, not their values, so reification doesn't change anything.